doc/map.md

```diff
@@ -0,0 +1,541 @@
+- `start_layer` and `num_layers` tell you which layers belong to this group. Obviously groups are not allowed to overlap
+- the 'Game' group, which is the only one that is allowed to hold physics layers, should have every field zeroed, only `x_parallax` and `y_parallax` should each be 100 and the `name` should be "Game"
+- all maps must have a 'Game' group, since every map must have a 'Game' layer which can only be in the 'Game' group
+
+Layers
+------
+
+- `type_id` = 5
+
+Layer types:
+
+- Tilemap layers:
+    - Tiles layer
+    - Physics layers:
+        - Game layer
+        - Front layer (DDNet only)
+        - Tele layer (DDNet only)
+        - Speedup layer (DDNet only)
+        - Switch layer (DDNet only)
+        - Tune layer (DDNet only)
+- Quads layer
+- Sounds layer (DDNet only)
+- Deprecated Sounds layer (DDNet only, replaced by Sounds layer)
+
+Note that:
+1. All physics layers *should* be unique, but this isn't properly enforced on all DDNet maps.
+Use the last physics layer of the type you seek.
+2. All maps must have a Game layer
+
+
+        item_data base for all layer items (different types have different extensions):
+            [1] _version (not used, was uninitialized)
+            [1] type
+            [1] flags
+
+- `flags` currently only has the detail flag (at 2^0), which is used in Quad-, Tile- and Sound layers.
+- `type` holds the type of layer:
+    - 2 -> Tilemap layer
+    - 3 -> Quads layer
+    - 9 -> Deprecated Sounds layer
+    - 10 -> Sounds layer
+
+
+    item_data extension for tilemap layers:
+        [1] version
+        [1] width
+        [1] height
+        [1] type
+        [4] color: Color
+        [1] opt *color_envelope
+        [1] color_envelope_offset
+        [1] opt *image
+        [1] &data: 2d-array of the the tile type 'Tile'
+        
+        version 3 extension:
+        [3] name: I32String
+        
+        DDNet extension (no version change):
+        [1] opt &data_tele
+        [1] opt &data_speedup
+        [1] opt &data_front
+        [1] opt &data_switch
+        [1] opt &data_tune
+
+- Vanilla is at `version` = 4, DDNet at `version` = 3
+- `width` and `height` specify the dimensions of the layer
+- `type` tells you what kind of tilemap layer this is:
+    - 0 -> Tiles
+    - 1 -> Game
+    - 2 -> Tele
+    - 4 -> Speedup
+    - 8 -> Front
+    - 16 -> Switch
+    - 32 -> Tune
+- `color`, `color_envelope`, `color_envelope_offset`, `image` are only used by the tiles layer
+
+Tile types:
+
+- all tile types consist of bytes (u8)
+- all tile types have the `id` byte, which identifies its use
+    - for example in the game layer, 0 is air, 1 is hookable, etc.
+- many have a `flags` byte, which is a bitflag with the following bits:
+    - 2^0 -> vertical flip
+    - 2^1 -> horizontal flip
+    - 2^2 -> opaque
+    - 2^3 -> rotate
+
+
+    'Tile' tile type (used by all vanilla layers and the front layer):
+        [1] id
+        [1] flags
+        [1] skip
+        [1] - unused
```

Comment:
A bit unintuitive that this starts to use bytes as unit, instead of integers as before.